demo.py

```diff
@@ -174,13 +175,16 @@ def copy_files(temp_folder: Path, output_dir: Path):
     setup_logging()
     logger.info("Starting up…")
 
-    for repo in tqdm(args.repositories, desc="Generating docs for repos"):
-        temp_repo_folder = create_temp_repo_folder()
-        try:
-            generate_docs_remote(repo, temp_repo_folder, local_dev=True)
+    with monitor_execution(run_id="demo_run", enabled=is_monitoring_enabled()) as mon:
```

Comment:
Can we improve the naming here, e.g:
```suggestion
    with monitor_execution(run_id="demo_run", enabled=is_monitoring_enabled()) as monitoring_context:
```